api/src/apps/deployment_unit.rs

```diff
@@ -484,4 +513,92 @@ mod tests {
             Some("postgres,wordpress,".to_string())
         );
     }
+
+    #[test]
+    fn should_apply_templating_on_merged_application_companions() {
+        let config = config_from_str!(
+            r#"
+            [companions.openid]
+            serviceName = 'openid'
+            type = 'application'
+            image = 'private.example.com/library/openid:latest'
+            env = [ "VAR_1={{application.name}}" ]
+        "#
+        );
+
+        let mut unit = DeploymentUnit::new(
+            AppName::from_str("master").unwrap(),
+            vec![sc!("openid", "private.example.com/library/openid:backup")],
+        );
+
+        unit.extend_with_config(&config).unwrap();
+
+        let openid_configs: Vec<_> = unit.try_into().unwrap();
+        assert_eq!(openid_configs.len(), 1);
+        let openid_env = openid_configs[0].env().unwrap();
+
+        assert_eq!(
```

Comment:
you could also assert that `"private.example.com/library/openid:backup"` is the service's image